src/wallet/lelantusjoinsplitbuilder.cpp

```diff
@@ -367,6 +368,17 @@ void LelantusJoinSplitBuilder::GenerateMints(const std::vector<CAmount>& newMint
         std::vector<unsigned char> encryptedValue = pwalletMain->EncryptMintAmount(mintVal, pubCoin.getValue());
 
         scriptSerializedCoin.insert(scriptSerializedCoin.end(), encryptedValue.begin(), encryptedValue.end());
+
+        auto pubcoin = hdMint.GetPubcoinValue() + lelantus::Params::get_default()->get_h1() * Scalar(hdMint.GetAmount()).negate();
+        uint256 hashPub = primitives::GetPubCoinValueHash(pubcoin);
+        CDataStream ss(SER_GETHASH, 0);
+        ss << hashPub;
+        ss << seedID;
+        uint256 hashForRecover = Hash(ss.begin(), ss.end());
+        CDataStream serializedHash(SER_GETHASH, 0);
```

Comment:
`SER_GETHASH` is misleading because you clearly _not_ serializing for hashing

Reply:
changed wrong one to SER_NETWORK. First serialization is done indeed for hash purposes, the second one is not.